CONTRIBUTING.md

```diff
@@ -289,6 +301,40 @@ do not decrease overall code coverage of the codebase - this is aligned with our
 goal to increase coverage over time. Keep track of execution time for your unit
 tests and try to keep them as short as possible.
 
+#### Testing Library Recommendations 
+
+To keep testing practices consistent across the project, it is advised to use these libraries under
+these circumstances:
+
+- For assertions to validate expectations, use `"github.com/stretchr/testify/assert"`
+- For assertions that are required to continue the test, use `"github.com/stretchr/testify/require"`
+- For mocking external resources, use `"github.com/stretchr/testify/mock"`
+- For validating HTTP traffic interactions, `"net/http/httptest"`
+
+### Integration Testing
+
+Integration testing is encouraged throughout the project, container images can be used in order to facilitate
+a local version. In their absence, it is strongly advised to mock the integration.
+
+### Using CGO
+
+Using CGO is prohibited due to the lack of portability and complexity 
+that comes with managing external libaries with different operating systems and configurations.
+However, if the package MUST use CGO, this should be explicitly called out within the readme
+with clear instructions on how to install the required libraries.
+Furthermore, if your package requires CGO, it MUST be able to compile and operate in a no op mode
+or report a warning back to the collector with a clear error saying CGO is required to work.
+
+### Configuration Naming
+
+In order to keep configuration settings consistent across the project and their intenet,
+a package must use the following naming for their purpose:
+
+| Name               | Purpose                                                              | Example                                                     |
+|--------------------|----------------------------------------------------------------------|-------------------------------------------------------------|
+| ExportTimeout      | Used to set the allowed during of an export operation                | Setting the allowed time a single export operation can take |
```

Comment:
This seems to be in response to #4497, which was discussed yesterday during the SIG meeting. Let's wait for the resolution before merging this.

Reply:
Yes, sorry I was meant to ping you on it but it had slipped my mind.

Reply:
In order to clear this PR for merging, would you please remove this, or add a link to the discussion? This doesn't seem to be settled yet.

Reply:
Done :)